src/BinaryParsers/PEBinary/ProgramDatabase/LinkerCommandLine.cs

```diff
@@ -0,0 +1,102 @@
+
+namespace Microsoft.CodeAnalysis.BinaryParsers.ProgramDatabase
+{
+    /// <summary>Processes command lines stored by linkers in PDBs.</summary>
+    internal struct LinkerCommandLine
+    {
+        /// <summary>
+        /// The raw, unadulterated command line before processing.
+        /// </summary>
+        public readonly string Raw;
+
+        /// <summary>
+        /// Whether or not this command line enables incremental linking
+        /// </summary>
+        public readonly bool IncrementalLinking;
+
+        /// <summary>
+        /// Initializes a new instance of the <see cref="LinkerCommandLine"/> struct from a raw PDB-supplied command line.
+        /// </summary>
+        /// <param name="commandLine">The raw command line from the PDB.</param>
+        public LinkerCommandLine(string commandLine)
+        {
+            this.Raw = commandLine ?? "";
+            this.IncrementalLinking = false;
+
+            // https://docs.microsoft.com/cpp/build/reference/incremental-link-incrementally?view=msvc-170
+            bool debugSet = false;
+            bool optRef = false;
+            bool optIcf = false;
+            bool optLbr = false;
+            bool order = false;
+            bool explicitlyEnabled = false;
+            bool explicitlyDisabled = false;
+            foreach (string argument in ArgumentSplitter.CommandLineToArgvW(commandLine))
+            {
+                if (!CommandLineHelper.IsCommandLineOption(argument))
+                {
+                    continue;
+                }
+
+                // There are multiple /debug options so use StartsWith
+                if (argument.StartsWith("/debug", System.StringComparison.OrdinalIgnoreCase))
+                {
+                    debugSet = true;
+                }
+                else if (string.Equals(argument, "/opt:ref", System.StringComparison.OrdinalIgnoreCase))
+                {
+                    optRef = true;
+                }
+                else if (string.Equals(argument, "/opt:icf", System.StringComparison.OrdinalIgnoreCase))
+                {
+                    optIcf = true;
+                }
+                else if (string.Equals(argument, "/opt:lbr", System.StringComparison.OrdinalIgnoreCase))
+                {
+                    optLbr = true;
+                }
+                else if (string.Equals(argument, "/order", System.StringComparison.OrdinalIgnoreCase))
+                {
+                    order = true;
+                }
+                else if (string.Equals(argument, "/incremental", System.StringComparison.OrdinalIgnoreCase))
+                {
+                    explicitlyEnabled = true;
+                }
+                else if (string.Equals(argument, "/incremental:no", System.StringComparison.OrdinalIgnoreCase))
+                {
+                    explicitlyDisabled = true;
+                }
+            }
+
+            // Explicit enable or disable wins.
```

Comment:
>Explicit [](http://example.com/codeflow?start=15&length=8)

This is a nice comment that provides some reassurance to me that we understand the command-line interpretation behavior, thanks for adding it.

Reply:
I think you have more confidence than I do!  

I don't work on MSVC and have no knowledge of the inner workings and am attempting to encode the public documentation (from [https://docs.microsoft.com/en-us/cpp/build/reference/incremental-link-incrementally?view=msvc-170](https://docs.microsoft.com/en-us/cpp/build/reference/incremental-link-incrementally?view=msvc-170)).   This comment is regarding the interpretation in this code.  It could be inaccurate to how link.exe behaves.

Reply:
Thanks for correcting my optics. :) I can probably locate relevant MSVC parsing logic for us.

Reply:
@michaelcfanning I just wanted to check in regarding whether you were able to locate the parsing logic for the linker command line.  I believe this is the only open issue remaining in this PR.  Thanks.

Reply:
I will go check. We don't have to block this immediate PR for that, we can update as necessary once checked in. I'd like to understand whether we need to update the arg parsing to support both the '-' and '/' prefixes first, though (see separate comment).